src/System.Linq/src/System/Linq/Partition.cs

```diff
@@ -35,16 +35,45 @@ internal interface IIListProvider<TElement> : IEnumerable<TElement>
         int GetCount(bool onlyIfCheap);
     }
 
+    /// <summary>
+    /// An iterator that supports random access and can produce a partial sequence of its items through an optimized path.
+    /// </summary>
     internal interface IPartition<TElement> : IIListProvider<TElement>
     {
+        /// <summary>
+        /// Skips the specified number of elements in this sequence.
```

Comment:
I think it'd help to avoid confusion if this were prefaced with "Creates a new partition that"... same for Take.  Otherwise it sounds to me like it's mutating this instance.